### PR TITLE
Btrfs filesystem create support

### PR DESCRIPTION
**Pull request by jelly**
Pixel test changes: https://cockpit-logs.us-east-1.linodeobjects.com/pull-19880-20240124-121638-97e94766-fedora-39-storage/log.html

We should probably only take pixel tests once?

### CODE REVIEW

**Comment by jelly** [comment]
TestStorageMounting.testMounting is broken everywhere, this needs work.

**Comment by mvollmer** [comment]
Let's make it so that `TestStorageUsed.testUsed` keeps using ext4. Or do we want to duplicate this as well for both ext4 and btrfs?

**Comment by jelly** [comment]
> Let's make it so that `TestStorageUsed.testUsed` keeps using ext4. Or do we want to duplicate this as well for both ext4 and btrfs?

Sure and saves me a pixel test update :)

**Comment by mvollmer** [comment]
@jelly, I think we also need btrfs filesystem deletion support if we offer creation. Otherwise people can create a btrfs filesystem in Cockpit, and can never get rid of it again in Cockpit itself.

**Comment by mvollmer** [comment]
> @jelly, I think we also need btrfs filesystem deletion support if we offer creation. Otherwise people can create a btrfs filesystem in Cockpit, and can never get rid of it again in Cockpit itself.

Oh, that was easy! :-) I like how the old code for tearing down a filesystem does the right thing for a one-device btrfs.

**Comment by mvollmer** [comment]
> # test/common/storagelib.py:140: unused method 'dialog_wait_title' (60% confidence)

Let's remove dialog_wait_title to make tox happy.

**Comment by mvollmer** [comment]
>     # test/common/storagelib.py:140: unused method 'dialog_wait_title' (60% confidence)

Let's remove dialog_wait_title to make tox happy.

**Review by mvollmer — DISMISSED** [review]
Nice!

**Review by mvollmer — APPROVED** [review]
Nice!

File: pkg/storaged/utils.js [hunk 1]
```diff
@@ -891,15 +891,18 @@ export function get_active_usage(client, path, top_action, child_action, is_temp
         // HACK: get_active_usage is used for mounting and formatting so we use the absence of the subvol argument
         // to figure out that we want to format this device.
         // This is separate from the if's below as we also always have to umount the filesystem.
-        if (btrfs_volume && !subvol) {
+
+        // We allow a btrfs volume with one device to be formatted as this
+        // looks the most like a normal filesystem use case.
+        if (btrfs_volume && btrfs_volume.data.num_devices !== 1 && !subvol) {
```
Comment (cockpituous) [inline pkg/storaged/utils.js:897]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19880-20240130-123304-85339f80-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/utils.js [hunk 2]
```diff
@@ -891,15 +891,18 @@
...
             usage.push({
                 level,
                 usage: 'btrfs-device',
                 block,
                 btrfs_volume,
                 location: btrfs_volume.data.label || btrfs_volume.data.uuid,
                 actions: get_actions(_("remove from btrfs volume")),
-                blocking: true
+                blocking: true,
```
Comment (cockpituous) [inline pkg/storaged/utils.js:905]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19880-20240130-123304-85339f80-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/utils.js (hunk 1 above)
Comment (cockpituous) [inline pkg/storaged/utils.js:897]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19880-20240130-143133-85339f80-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/utils.js [hunk 3]
```diff
@@ -891,15 +891,18 @@ export function get_active_usage(client, path, top_action, child_action, is_temp
...
```
Comment (cockpituous) [inline pkg/storaged/utils.js:905]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19880-20240130-143133-85339f80-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Pull request merged**